app/models/workers_scaling/experiment_resources_interface.rb

```diff
@@ -22,26 +31,39 @@ def get_available_infrastructures
           .flat_map {|x| x.has_key?(:children) ? x[:children] : x }
           .select {|x| x[:enabled]}
           .map {|x| x[:infrastructure_name].to_sym}
-          .map do |infrastructure_name|
+          .flat_map do |infrastructure_name|
         InfrastructureFacadeFactory.get_facade_for(infrastructure_name).get_subinfrastructures(@user_id)
-      end.flatten
+      end
+    end
+
+    ##
+    # Returns amount of Workers that can be yet scheduled on given infrastructure
+    def current_infrastructure_limit(infrastructure)
+      infrastructure_limit = @infrastructure_limits.detect { |entry| entry[:infrastructure] == infrastructure }
+      if infrastructure_limit.nil?
+        Float::INFINITY
```

Comment:
Jeżeli ma być takie założenie to nazwałbym to allowed infrastructeres, a nie infrastructures limits. Dla mnie intuicyjne jest że jeśli nie podamy limitu to go nie ma, zwłaszcza jeśli odnosi się to do istnienia wpisu w liście limitów.